spark/src/main/java/org/apache/iceberg/spark/SparkTypeToType.java

```diff
@@ -46,12 +46,19 @@
 class SparkTypeToType extends SparkTypeVisitor<Type> {
   private final StructType root;
   private int nextId = 0;
+  private final boolean useTimestampWithoutZone;
```

Comment:
I'm not sure what this flag is for, I think it's probably safer to always return timestamptype.withZone from here, and handle the mismatch outside of this code

Reply:
Moved this logic to [SparkFixupTimestampType.java](https://github.com/sshkvar/iceberg/blob/f2fdf2a5f9ffc64b44b353e9fd7d09ed21f8c416/spark/src/main/java/org/apache/iceberg/spark/SparkFixupTimestampType.java)